drivers/timer/nrf_rtc_timer.c

```diff
@@ -189,8 +206,160 @@ void _timer_idle_enter(s32_t sys_ticks)
+
+/**
+ * @brief provides total systicks programed.
+ *
+ * returns : total number of sys ticks programed.
+ */
+
+uint32_t _get_program_time(void)
+{
+	return expected_sys_ticks;
+}
+
+/**
+ * @brief provides total systicks remaining since last programing of RTC.
+ *
+ * returns : total number of sys ticks remaining since last RTC programing.
+ */
+
+uint32_t _get_remaining_program_time(void)
+{
+	u64_t rtc_elapsed;
+	u32_t rtc_now;
+
+	if (!expected_sys_ticks) {
+		return 0;
+	}
+
+	rtc_now = RTC_COUNTER;
+
+	/* Number of RTC cycles passed since last RTC Programing*/
+	rtc_elapsed = (rtc_now - rtc_past) & RTC_MASK;
+
+	return (expected_sys_ticks - (rtc_elapsed / RTC_TICKS_PER_SYS_TICK));
+}
+
+/**
+ * @brief provides total systicks passed since last programing of RTC.
+ *
+ * returns : total number of sys ticks passed since last RTC programing.
+ */
+
+uint32_t _get_elapsed_program_time(void)
+{
+	u32_t rtc_now, rtc_elapsed;
+
+	if (!expected_sys_ticks) {
+		return 0;
+	}
+
+	rtc_now = RTC_COUNTER;
+
+	/* Number of RTC cycles passed since last RTC Programing*/
+	rtc_elapsed = (rtc_now - rtc_past) & RTC_MASK;
+
+	/*Convert number of Machine cycles to SYS TICS*/
+	return (rtc_elapsed / RTC_TICKS_PER_SYS_TICK);
+}
+
+
+/**
+ * @brief Sets interrupt for rtc compare value for systick time.
+ *
+ * This Function does following:-
+ * 1. Updates expected_sys_ticks equal to time.
+ * 2. Update kernel book keeping for time passed since device bootup.
+ * 3. Calls routine to set rtc interrupt.
+ */
+
+void _set_time(uint32_t time)
+{
+	if (!time) {
+		expected_sys_ticks = 0;
+		return;
+	}
+
+	/* Update expected_sys_ticls to time to programe*/
+	expected_sys_ticks = time;
+
+	_sys_clock_tick_count = _get_elapsed_clock_time();
+	/* Programe RTC compare register to generate interrupt*/
+	rtc_compare_set(rtc_past +
+			(expected_sys_ticks * RTC_TICKS_PER_SYS_TICK));
+}
+
+/**
+ * @brief provides time remaining to reach rtc count overflow.
+ *
+ * This function returns how many sys RTC remaining for rtc to overflow.
+ * This will be required when we will programe RTC compare value to maximum
+ * possible value.
+ *
+ * returns : difference between current systick and Maximum possible systick.
+ */
+int32_t _get_max_clock_time(void)
+{
+	u32_t rtc_now, rtc_away, sys_away = 0;
+
+	rtc_now = RTC_COUNTER;
+	rtc_away = (RTC_MASK - rtc_now);
+
+	if (rtc_away >= RTC_TICKS_PER_SYS_TICK) {
+		sys_away = rtc_away / RTC_TICKS_PER_SYS_TICK;
+	}
+	return sys_away;
+}
+
+/**
+ * @brief Enable sys Clock.
+ *
+ * This is used to programe RTC clock to maximum Clock time incase Clock to
+ * remain On.
+ */
+void _enable_sys_clock(void)
+{
+	if (!expected_sys_ticks) {
+		/* Programe sys tick to Maximum possible value*/
+		_set_time(_get_max_clock_time());
+	}
+}
+
+/**
+ * @brief provides total systicks passed since device bootup.
+ *
+ * returns : total number of sys ticks passed since device bootup.
+ */
+
+uint64_t _get_elapsed_clock_time(void)
```

Comment:
this function, frankly needs more understand on my side. Others to review  here.